benchmarks/benchmarks/generate_data/um_files.py

```diff
@@ -0,0 +1,212 @@
+        new_field.bzy = new_ff.real_constants.start_lat - 0.5 * new_field.bdy
+
+        # Hemisphere.
+        new_field.lbhem = 32
+        # Processing.
+        new_field.lbproc = 0
+
+        # Vertical.
+        # Hybrid height values by simulating sequences similar to those in a
+        #  theta file.
+        new_field.lbvc = 65
+        if level_ == 0:
+            new_field.lblev = 9999
+        else:
+            new_field.lblev = level_
+
+        level_1 = level_ + 1
+        six_rec = 20 / 3
+        three_rec = six_rec / 2
+
+        new_field.blev = level_1 ** 2 * six_rec - six_rec
+        new_field.brsvd1 = (
+            level_1 ** 2 * six_rec + (six_rec * level_1) - three_rec
+        )
+
+        brsvd2_simulated = np.linspace(0.995, 0, len_z)
+        shift = min(len_z, 2)
+        bhrlev_simulated = np.concatenate(
+            [np.ones(shift), brsvd2_simulated[:-shift]]
+        )
+        new_field.brsvd2 = brsvd2_simulated[level_]
+        new_field.bhrlev = bhrlev_simulated[level_]
+
+        # Time.
+        new_field.lbtim = 11
+
+        new_field.lbyr = time_step_
+        for attr_name in ["lbmon", "lbdat", "lbhr", "lbmin", "lbsec"]:
+            setattr(new_field, attr_name, 0)
+
+        new_field.lbyrd = time_step_ + 1
+        for attr_name in ["lbmond", "lbdatd", "lbhrd", "lbmind", "lbsecd"]:
+            setattr(new_field, attr_name, 0)
+
+        # Data and packing.
+        new_field.lbuser1 = 1
+        new_field.lbpack = int(compress)
+        new_field.bacc = 0
+        new_field.bmdi = -1
+        new_field.lbext = 0
+        new_field.set_data_provider(array_provider)
+
+        new_ff.fields.append(new_field)
+
+    for time_step in range(len_t):
+        for level in range(len_z):
+            add_field(level, time_step + 1)
+
+    ff_path = save_paths.get("FF", None)
+    pp_path = save_paths.get("PP", None)
+    nc_path = save_paths.get("NetCDF", None)
+
+    if ff_path:
+        new_ff.to_file(ff_path)
+    if pp_path:
+        fields_to_pp_file(str(pp_path), new_ff.fields)
+    if nc_path:
+        temp_ff_path = None
+        # Need an Iris Cube from the FF content.
+        if ff_path:
+            # Use the existing file.
+            ff_cube = load_cube(ff_path)
+        else:
+            # Make a temporary file.
+            temp_ff_path = NamedTemporaryFile()
+            new_ff.to_file(temp_ff_path.name)
+            ff_cube = load_cube(temp_ff_path.name)
+
+        save_cube(ff_cube, nc_path, zlib=compress)
+        if temp_ff_path:
+            temp_ff_path.close()
+
+
+FILE_EXTENSIONS = {"FF": "", "PP": ".pp", "NetCDF": ".nc"}
+
+
+def create_um_files(
+    len_x: int,
+    len_y: int,
+    len_z: int,
+    len_t: int,
+    compress: bool,
+    file_types: list,
+) -> dict:
+    """
+    Generate FF-based FF / PP / NetCDF files with specified shape and compression.
+
+    Saved to a directory for all files that shape. A dictionary of the saved
```

Comment:
Bit of a glitch in the descrtiption here:
>  "all files that shape".

Reply:
9fbcb336393e19c6d2ed3146b14509eb5808e374